RestartWindow.qml

```diff
@@ -753,27 +812,124 @@ Popup {
+                            opacity: hidden_opacity
+                        }
+                    }
+
+                    Rectangle {
+                        color: 'black'
+                        border.color: grid_item_border_color
+                        border.width: grid_item_border_width
+                        Layout.row: 4
+                        Layout.column: 0
+                        Layout.fillWidth: true
+                        Layout.minimumWidth: (mcu_check.width / 4) - 1
+                        Layout.preferredWidth: mcu_check.width / 4
+                        Layout.preferredHeight: mcu_check_pref_height
+
+                        Text {
+                            id: charge_enable_text
+
+                            property bool charge_enable_fault: backEnd.charge_enable && backEnd.restart_enable
+                            onCharge_enable_faultChanged: {
+                                if(charge_enable_fault) {
+                                    this.opacity = 1;
+                                }
+                            }
+
+                            anchors.centerIn: parent
+                            text: "Charge enable"
+                            color: charge_enable_fault ? "red" : "white"
                             font.pointSize: mcu_check_point_size
                             opacity: hidden_opacity
                         }
                     }
 
                     Rectangle {
-                        id: cell_group_fault_rect
                         color: 'black'
+                        border.color: grid_item_border_color
+                        border.width: grid_item_border_width
                         Layout.row: 4
+                        Layout.column: 1
+                        Layout.fillWidth: true
+                        Layout.minimumWidth: (mcu_check.width / 4) - 1
+                        Layout.preferredWidth: mcu_check.width / 4
+                        Layout.preferredHeight: mcu_check_pref_height
+
+                        Text {
+                            id: mc_status_text
+
+                            property int mc_status_fault: backEnd.mc_status
+                            property var err: ["", "Over Current", "Unused", "Hall Sensor fault", "Motor Locked", "Sensor Fault1", "Sensor Fault2", "Unused",
+                                            "High Battery Voltage", "Controller Over Heat"]
+                            onMc_status_faultChanged: {
+                                if(mc_status_fault > 0 && backEnd.restart_enable) {
+                                    if(mc_status_fault == 9  && backEnd.motor_controller_temp < 105) {
+                                        mc_status_fault = false;
+                                    } else {
+                                        this.opacity = 1;
+                                    }
+                                    console.log("mc_stat: " + mc_status_fault +err[mc_status_fault])
```

Comment:
```suggestion
                            property int mc_status_fault: (backEnd.mc_status == 9 && backEnd.motor_controller_temp < 105) ? 0 : backEnd.mc_status
                            property var err: ["", "Over Current", "Unused", "Hall Sensor fault", "Motor Locked", "Sensor Fault 1", "Sensor Fault 2", "Unused",
                                            "High Battery Voltage", "Controller Over Heat"]
                            onMc_status_faultChanged: {
                                if(mc_status_fault > 0 && backEnd.restart_enable) {
                                    this.opacity = 1;
```
For some reason, it needs to be refactored this way to work properly (I think it doesn't like setting `mc_status_fault` outside of the property definition.